src/pages/workspaces/workspace/RequestAccessModal.js

```diff
@@ -0,0 +1,103 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { div, h, table, tbody, td, th, thead, tr } from 'react-hyperscript-helpers'
+import { buttonPrimary } from 'src/components/common'
+import { centeredSpinner } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { cond, useOnMount, withBusyState } from 'src/libs/utils'
+
+
+export const RequestAccessModal = ({ onDismiss, workspace }) => {
+  const [groups, setGroups] = useState([])
+  const [accessInstructions, setAccessInstructions] = useState([])
+  const [loadingGroups, setLoadingGroups] = useState(false)
+  const [loadingAccessInstructions, setLoadingAccessInstructions] = useState(false)
+  const signal = useCancellation()
+
+  const { Groups, Workspaces } = Ajax(signal)
+
+  const fetchGroups = _.flow(
+    withBusyState(setLoadingGroups),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    setGroups(await Groups.list())
+  })
+
+  const fetchAccessInstructions = _.flow(
+    withBusyState(setLoadingAccessInstructions),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    const i = await Workspaces.workspace(workspace.workspace.namespace, workspace.workspace.name).accessInstructions()
+    console.log(i)
+    setAccessInstructions(i)
+  })
+
+  useOnMount(() => {
+    fetchGroups()
+    fetchAccessInstructions()
```

Comment:
I considered, and I definitely experimented with, code like that (which might be why I ended up with the same error message for both). Looking again, I think I can see how it would work so I'll do a short time-box experiment to see how it turns out.

Reply:
Updated. Let me know what you think. Thanks!